drivers/SmartThings/zigbee-air-quality-detector/src/MultiIR/init.lua

```diff
@@ -0,0 +1,182 @@
+
+local MultiIR_SENSOR_FINGERPRINTS = {
+  { mfr = "MultiIR", model = "PMT1006S-SGM-ZTN" }--This is not a sleep end device
+}
+
+local function can_handle_MultiIR_sensor(opts, driver, device)
+  for _, fingerprint in ipairs(MultiIR_SENSOR_FINGERPRINTS) do
+    if device:get_manufacturer() == fingerprint.mfr and device:get_model() == fingerprint.model then
+      return true
+    end
+  end
+  return false
+end
+
+local function send_read_attr_request(device, cluster, attr)
+  device:send(
+    cluster_base.read_manufacturer_specific_attribute(
+      device,
+      cluster.id,
+      attr.id,
+      cluster.mfg_specific_code
+    )
+  )
+end
+
+local function do_refresh(driver, device)
+  device:send(RelativeHumidity.attributes.MeasuredValue:read(device):to_endpoint(0x01))
+  device:send(TemperatureMeasurement.attributes.MeasuredValue:read(device):to_endpoint(0x01))
+  send_read_attr_request(device, custom_clusters.particulate_matter, custom_clusters.particulate_matter.attributes.pm2_5_MeasuredValue)
+  send_read_attr_request(device, custom_clusters.particulate_matter, custom_clusters.particulate_matter.attributes.pm1_0_MeasuredValue)
+  send_read_attr_request(device, custom_clusters.particulate_matter, custom_clusters.particulate_matter.attributes.pm10_MeasuredValue)
+  send_read_attr_request(device, custom_clusters.unhealthy_gas, custom_clusters.unhealthy_gas.attributes.CH2O_MeasuredValue)
+  send_read_attr_request(device, custom_clusters.unhealthy_gas, custom_clusters.unhealthy_gas.attributes.tvoc_MeasuredValue)
+  send_read_attr_request(device, custom_clusters.carbonDioxide, custom_clusters.carbonDioxide.attributes.measured_value)
+end
+
+local LEVEL_TO_NUMBER = {
+  { level = "good", value = 1 },
+  { level = "moderate", value = 2 },
+  { level = "unhealthy", value = 3 }
+}
+
+local cap_index = {
+  carbonDioxide = 1,
+  pm2_5 = 2,
+  pm1_0 = 3,
+  pm10 = 4,
+  tvoc = 5
+}
+
+local cap_save_level = {
+  { level = "good", value = 1 },
+  { level = "good", value = 1 },
+  { level = "good", value = 1 },
+  { level = "good", value = 1 },
+  { level = "good", value = 1 }
+}
+
+local function airQualityHealthConcern_handler(device,ep,index,level)
+  --local airQuality_level = "unhealthy"
+  local max = 1
+  local max_cap = 3
+  --level is converted to a number and save it
+  for _,value in ipairs(LEVEL_TO_NUMBER) do
+    if level == value.level then
+      cap_save_level[index].value = value.value
+      cap_save_level[index].level = level
+    end
+  end
+  --Find the maximal of all levels
+  for i=1,5 do
+    if cap_save_level[i].value > max then
+      max = cap_save_level[i].value;
+      max_cap = i
+    end
+  end
+  --The maximum value represents the air quality class
+  local airQuality_level = cap_save_level[max_cap].level
+  device:emit_event_for_endpoint(ep, capabilities.airQualityHealthConcern.airQualityHealthConcern({value = airQuality_level}))
+end
+
+local function carbonDioxide_attr_handler()
```

Comment:
There are a couple of places where you've created these functions, I think intending to use them as generic factories, and then only used them once. In these cases it would be clearer to just make them explicit functions.